libs/sdk-ui-pivot/src/columnWidths.ts

```diff
@@ -100,68 +113,250 @@ export type ColumnWidthItem =
+export function isWeakMeasureColumnWidthItem(obj: unknown): obj is IWeakMeasureColumnWidthItem {
     return (
-        !isEmpty(columnWidthItem) &&
-        (columnWidthItem as IWeakMeasureColumnWidthItem).measureColumnWidthItem !== undefined &&
-        (columnWidthItem as IWeakMeasureColumnWidthItem).measureColumnWidthItem.locator !== undefined
+        !isEmpty(obj) &&
+        (obj as IWeakMeasureColumnWidthItem).measureColumnWidthItem !== undefined &&
+        (obj as IWeakMeasureColumnWidthItem).measureColumnWidthItem.locator !== undefined
     );
 }
 
-// TODO: add factory functions
+function newMeasureColumnLocator(measureOrId: IMeasure | string): IMeasureColumnLocator {
+    const measureIdentifier = measureLocalId(measureOrId);
+
+    return {
+        measureLocatorItem: {
+            measureIdentifier,
+        },
+    };
+}
+
+/**
+ * Sets width of a column which contains values of a row attribute.
+ *
+ * @param attributeOrId - attribute specified by value or by localId reference
+ * @param width - width in pixels
+ * @param allowGrowToFit - indicates whether the column is allowed to grow if the table's growToFit is enabled
+ * @public
+ */
+export function widthForAttributeColumn(
+    attributeOrId: IAttribute | string,
+    width: number,
+    allowGrowToFit?: boolean,
+): IAttributeColumnWidthItem {
+    return {
+        attributeColumnWidthItem: {
+            attributeIdentifier: attributeLocalId(attributeOrId),
+            width: {
+                value: width,
+                allowGrowToFit,
+            },
+        },
+    };
+}
+
+/**
+ * Sets width for all measure columns in the table.
+ *
+ * @param width - width in pixels
+ * @param allowGrowToFit - indicates whether the column is allowed to grow if the table's growToFit is enabled
+ * @public
+ */
+export function widthForAllMeasureColumns(
+    width: number,
+    allowGrowToFit?: boolean,
+): IAllMeasureColumnWidthItem {
+    return {
+        measureColumnWidthItem: {
+            width: {
+                value: width,
+                allowGrowToFit,
+            },
+        },
+    };
+}
+
+/**
+ * Sets width for all columns containing values of the provided measure.
+ *
+ * @param measureOrId - measure specified either by value or by localId reference
+ * @param width - width in pixels
+ * @param allowGrowToFit - indicates whether the column is allowed to grow if the table's growToFit is enabled
+ * @public
+ */
+export function widthForAllColumnsForMeasure(
+    measureOrId: IMeasure | string,
+    width: number,
+    allowGrowToFit?: boolean,
+): IWeakMeasureColumnWidthItem {
+    const locator = newMeasureColumnLocator(measureOrId);
+
+    return {
+        measureColumnWidthItem: {
+            width: {
+                value: width,
+                allowGrowToFit,
+            },
+            locator,
+        },
+    };
+}
+
+/**
+ * Sets width for all columns containing values of the provided measure. See also {@link newAttributeColumnLocator} to
```

Comment:
This is a bit misleading, this does not really set anything, it returns a new object for the setting, right? But I can't think of a better name: `newWidthForSelectedColumns` is kind of weird...